qcengine/programs/qcore.py

```diff
@@ -0,0 +1,275 @@
+            for shell in center["electron_shells"]:
+                shell.pop("normalized_primitives", None)
+                for el_k in ["coefficients", "exponents", "angular_momentum"]:
+                    shell[el_k] = shell[el_k].tolist()
+
+            if center["ecp_potentials"] is not None:
+                for shell in center["ecp_potentials"]:
+                    shell.pop("ecp_potentials", None)
+                    for ecp_k in ["angular_momentum", "r_exponents", "gaussian_exponents", "coefficients"]:
+                        shell[ecp_k] = shell[ecp_k].tolist()
+
+        basis_set = BasisSet(
+            name=str(input_model.model.basis), center_data=obas["center_data"], atom_map=obas["atom_map"]
+        )
+
+        wavefunction = {"basis": basis_set}
+        for key, qcschema_key in wavefunction_map.items():
+            qcore_data = output["wavefunction"].get(key, None)
+            if qcore_data is None:
+                continue
+
+            if ("density" in key) or ("fock" in key):
+                qcore_data = reorder_row_and_column_ao_indices(qcore_data, basis_set, self._qcore_to_cca_ao_order)
+            # Handles orbitals and 1D
+            elif "orbitals" in key:
+                qcore_data = reorder_column_ao_indices(qcore_data, basis_set, self._qcore_to_cca_ao_order)
+            elif "eigenvalues" in key:
+                qcore_data = reorder_column_ao_indices(
+                    qcore_data.reshape(1, -1), basis_set, self._qcore_to_cca_ao_order
+                ).ravel()
+
+            elif "occupations" in key:
+                tmp = np.zeros(basis_set.nbf)
+                tmp[: qcore_data.shape[0]] = qcore_data
+                qcore_data = reorder_column_ao_indices(
+                    tmp.reshape(1, -1), basis_set, self._qcore_to_cca_ao_order
+                ).ravel()
+            else:
+                raise KeyError("Wavefunction conversion key not understood")
+
+            wavefunction[qcschema_key] = qcore_data
+
+        wavefunction["restricted"] = True
+        if "scf_eigenvalues_b" in wavefunction:
+            wavefunction["restricted"] = False
+
+        output_data["wavefunction"] = wavefunction
+
+        # Handle remaining top level keys
+        properties = {
+            "calcinfo_nbasis": basis_set.nbf,
+            "calcinfo_nmo": basis_set.nbf,
+            "calcinfo_nalpha": np.sum(wavefunction["scf_occupations_a"] > 0),
+            "calcinfo_natom": input_model.molecule.symbols.shape[0],
+            "return_energy": output["energy"],
+        }
+        if wavefunction["restricted"]:
+            properties["calcinfo_nbeta"] = properties["calcinfo_nalpha"]
+        else:
+            properties["calcinfo_nbeta"] = np.sum(wavefunction["scf_occupations_b"] > 0)
+
+        output_data["properties"] = properties
+
+        output_data["schema_name"] = "qcschema_output"
+        output_data["success"] = True
+
+        return AtomicResult(**output_data)
```

Comment:
anything special needed to merge input and result extras field?

Reply:
Shouldn't be.